src/Tribe/Commerce/PayPal/Custom_Argument.php

```diff
@@ -77,19 +77,20 @@ public static function decode( $encoded, $assoc_array = false ) {
 			$encoded = str_replace( '\"', '"', $encoded );
 		}
 
-
 		// in case we receive the param in non json 'user_id:0,tribe_handler:tpp,pid:4' format
-		if ( false === $decoded = json_decode( urldecode_deep( $encoded ), $assoc_array ) ) {
+		if ( null === json_decode( urldecode_deep( $encoded ), $assoc_array ) ) {
 			// we create an array that string
 			$encoded = explode( ',', $encoded );
 
 			// Set the proper keys and values for the new array
-			$encoded = array_reduce( array( __CLASS__, 'array_fix_keys' ), array() );
+			$encoded = array_reduce( $encoded, array( __CLASS__, 'array_fix_keys' ), array() );
 
 			// we convert it into a json
 			$encoded = json_encode( $encoded );
 		}
 
+		$decoded = json_decode( urldecode_deep( $encoded ), $assoc_array );
```

Comment:
In the event that the initial `json_decode()` is not null, we'd end up doing `json_decode()` twice. Perhaps something like this would be more efficient?

```php
if ( null === $decoded = json_decode( urldecode_deep( $encoded ), $assoc_array ) ) {
  // do stuff with $encoded

  $decoded = json_decode( urldecode_deep( $encoded ), $assoc_array );
}
```

Reply:
The first `json_decode` is to check if it's a valid JSON, and if not (that's the genesis of this issue, PayPal changing the way they send back the data)we have to recreate it from the string we're receiving (i.e.:`user_id:0,tribe_handler:tpp,pid:4'`).

The second `json_decode` is to actually work around with the valid JSON (if it comes as a valid parameter or if we recreate it from what we get).

If I move the second one inside the IF, then if they send a valid JSON, the function won't be returning the decoded value.

Reply:
Already worked on something like that after the first CR here: https://github.com/moderntribe/event-tickets/pull/780/files and broke the patch